src/interceptors/XMLHttpRequest/XMLHttpRequestController.ts

```diff
@@ -305,10 +305,11 @@ export class XMLHttpRequestController {
 
     const totalResponseBodyLength = response.headers.has('Content-Length')
       ? Number(response.headers.get('Content-Length'))
-      : /**
-         * @todo Infer the response body length from the response body.
-         */
-        undefined
+      : (async () => {
```

Comment:
We should move this to a standalone function and await it here. I've just merged #613 that makes `respondWith` asynchronous, which will allow us this. 

The reasoning is that reading the response body is async. So this method will continue dispatching `loadstart` and other events while the body is being read, having `total` as `0` still.

Reply:
This is also why we need tests for this.

Reply:
Can you help me with tests

Reply:
Sure thing. I will add some next week.